common/src/main/scala/org/neo4j/spark/util/Neo4jUtil.scala

```diff
@@ -239,46 +242,111 @@ object Neo4jUtil {
 
   def connectorVersion: String = properties.getOrDefault("version", "UNKNOWN").toString
 
-  def valueToCypherExpression(value: Any): Expression = {
+  def getCorrectProperty(container: PropertyContainer, attribute: String): Property = {
+    container.property(attribute.split('.'): _*)
+  }
+
+  @tailrec
+  def getAttributeAndValue(filter: Filter): Seq[Any] = {
+    filter match {
+      case f: EqualTo => Seq(createParameterName(f.attribute, f.value), f.value)
+      case f: GreaterThan => Seq(createParameterName(f.attribute, f.value), f.value)
+      case f: GreaterThanOrEqual => Seq(createParameterName(f.attribute, f.value), f.value)
+      case f: LessThan => Seq(createParameterName(f.attribute, f.value), f.value)
+      case f: LessThanOrEqual => Seq(createParameterName(f.attribute, f.value), f.value)
+      case f: In => Seq(createParameterName(f.attribute, f.values), f.values)
+      case f: StringStartsWith => Seq(createParameterName(f.attribute, f.value), f.value)
+      case f: StringEndsWith => Seq(createParameterName(f.attribute, f.value), f.value)
+      case f: StringContains => Seq(createParameterName(f.attribute, f.value), f.value)
+      case f: Not => getAttributeAndValue(f.child)
+      case _ => Seq()
+    }
+  }
```

Comment:
maybe it's better to transform to an implicit function

Reply:
you mean the `createParameterName` ?

Reply:
I meant both!